packages/google_sign_in/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -345,10 +354,11 @@ public void init(
                 .getIdentifier("default_web_client_id", "string", context.getPackageName());
         if (!Strings.isNullOrEmpty(clientId)) {
           optionsBuilder.requestIdToken(clientId);
-          optionsBuilder.requestServerAuthCode(clientId);
+          optionsBuilder.requestServerAuthCode(clientId, forceCodeForRefreshToken);
         } else if (clientIdIdentifier != 0) {
           optionsBuilder.requestIdToken(context.getString(clientIdIdentifier));
-          optionsBuilder.requestServerAuthCode(context.getString(clientIdIdentifier));
+          optionsBuilder.requestServerAuthCode(
+              context.getString(clientIdIdentifier), forceCodeForRefreshToken);
```

Comment:
There's no testing of either of these changes, which are really the core of the change; if all of the Java part of this PR were reverted the new tests here would still pass. The tests you have are good for the Dart side, but there needs to be coverage of this (either via integration tests, or more likely via Java unit tests).

Reply:
Do you have a suggestion on the best way to go about that? From what I can tell, there aren't really any tests around the calls to `optionsBuilder`. The closest is checking the `requestedScopes` which is a property on the Delegate class. There isn't actually any logic to it on the java side, it just passes through the value from dart.

Reply:
> Do you have a suggestion on the best way to go about that?

You could refactor slightly so that the `optionsBuilder` instance comes from a replaceable method instead of just being locally declared. Then a unit test could set a mock `GoogleSignInOptions` instance, and you could validate calls to it.

> There isn't actually any logic to it on the java side, it just passes through the value from dart.

I'm not sure what you mean by "no logic". Is your assertion that nobody could cause this feature to completely stop working without any of the tests here failing, or do you think that nobody would ever make a change that accidentally broke it?

If it's the latter, that's really not a good bet. People accidentally break things when making other changes all the time. And in fact, I've seen changes where *the submitted PR* did not actually work even at the time because the author made a mistake sometime between their last manual test and the final uploaded version.

Reply:
All right, I took a crack at the refactoring. Let me know how that looks.